Add ASA Users documentation (#1510)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/asa/users/index.md

---
title: ASA Users
category: asa
---

# ASA Users API

> ## Get Started

Reviewer: ```suggestion
## Get started
```

| Product  | API Basics  | API Namespace        |
|----------|-------------|----------------------|
| [Advanced Server Access](https://www.okta.com/products/advanced-server-access/) | [How the ASA API works](../introduction/) | `https://app.scaleft.com/v1/`

> An Advanced Server Access (ASA) User corresponds to a human or service user in the ASA ecosystem.

Reviewer: ```suggestion
An Advanced Server Access (ASA) User corresponds to a human or Service User in the ASA ecosystem.
```

Explore the Users API: [![Run in Postman](https://run.pstmn.io/button.svg)](https://www.getpostman.com/run-collection/fba803e43a4ae53667d4).

> ## Users API Operations

Reviewer: ```suggestion
## Users API operations
```

The Users API has the following operations:
* [List the ASA Users for an ASA Team](#list-the-asa-users-for-an-asa-team)
* [Fetch an ASA User](#fetch-an-asa-user)
* [Update an ASA User](#update-an-asa-user)

> * [List ASA Groups a specific ASA User is a member of](#list-asa-groups-a-specific-asa-user-is-a-member-of)

Reviewer: ```suggestion
* [List ASA Groups that a specific ASA User is a member of](#list-asa-groups-a-specific-asa-user-is-a-member-of)
```

### List the ASA Users for an ASA Team

<ApiOperation method="GET" url="https://app.scaleft.com/v1/teams/${team_name}/users" />

> This endpoint list the ASA Users for an ASA Team.

Reviewer: ```suggestion
Lists the ASA Users for an ASA Team
```

> This endpoint requires one of the following roles: `access_user`, `access_admin`, `reporting_user`.

Reviewer: ```suggestion
This endpoint requires one of the following roles: `access_user`, `access_admin`, or `reporting_user`.
```

				"full_name": "Jason Compson IV",
				"last_name": "Compson"
			},
			"id": "5b9d691f-7207-47a6-9395-a8bf1b923c5a",
			"name": "Jason.Compson.IV",
			"oauth_client_application_id": null,
			"role_grants": null,
			"status": "ACTIVE",
			"user_type": "human"
		},
		{
			"deleted_at": null,
			"details": {
				"email": "benjy.compson@example.com",
				"first_name": "Benjy",
				"full_name": "Benjy Compson",
				"last_name": "Compson"
			},
			"id": "394a16e4-8f50-4017-9e98-b6e61ad96ca0",
			"name": "Benjy.Compson",
			"oauth_client_application_id": null,
			"role_grants": null,
			"status": "DISABLED",
			"user_type": "human"
		},
		{
			"deleted_at": "1910-06-10T00:00:00Z",
			"details": {
				"email": "quentin.compson@example.com",
				"first_name": "Quentin",
				"full_name": "Quentin Compson III",
				"last_name": "Compson"
			},
			"id": "c700f6c5-3ed0-448a-908d-e9bc939ac818",
			"name": "Quentin.Compson.III",
			"oauth_client_application_id": null,
			"role_grants": null,
			"status": "DELETED",
			"user_type": "human"
		}
	]
}
```
### Fetch an ASA User

<ApiOperation method="GET" url="https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}" />

> This endpoint fetches information regarding a specific ASA User.

Reviewer: ```suggestion
Fetches information about a specific ASA User
```

> This endpoint requires one of the following roles: `reporting_user`, `access_user`, `access_admin`.

Reviewer: ```suggestion
This endpoint requires one of the following roles: `reporting_user`, `access_user`, or `access_admin`.
```

| `id`   | string | The UUID of the ASA User. |
| `name`   | string | The username of the ASA User. |
| `oauth_client_application_id`   | string | The ID of the ASA User provided by OAuth, if it exists. |
| `role_grants`   | array | The permission roles available to the ASA User. |
| `status`   | string | One of `ACTIVE`, `DISABLED`, or `DELETED`. ASA Users cannot disable or delete their own ASA User. |
| `user_type`   | string | The type of ASA User, either `service` or `human`. |

#### Usage example

##### Request

```bash
curl -v -X GET \
-H "Authorization: Bearer ${jwt}" \
https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}
```

##### Response

```json
{
	"deleted_at": null,
	"details": {
		"email": "jason.compson@example.com",
		"first_name": "Jason",
		"full_name": "Jason Compson IV",
		"last_name": "Compson"
	},
	"id": "5b9d691f-7207-47a6-9395-a8bf1b923c5a",
	"name": "Jason.Compson.IV",
	"oauth_client_application_id": null,
	"role_grants": null,
	"status": "ACTIVE",
	"user_type": "human"
}
```
### Update an ASA User

<ApiOperation method="PUT" url="https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}" />

> This endpoint updates a specific ASA User.

Reviewer: ```suggestion
Updates a specific ASA User
```

> This endpoint requires one of the following roles: `access_admin`.

Reviewer: ```suggestion
This endpoint requires the `access_admin` role.
```

#### Request path parameters

| Parameter | Type        | Description   |
| --------- | ----------- | ------------- |
| `team_name`   | string | The name of your Team. |
| `user_name`   | string | The relevant username. |

#### Request query parameters

This endpoint has no query parameters.

#### Request body

This endpoint has no request body.

#### Response body
This endpoint returns a `204 No Content` response on a successful call.

#### Usage example

##### Request

```bash
curl -v -X PUT \
-H "Authorization: Bearer ${jwt}" \
https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}
```

##### Response

```json
HTTP 204 No Content
```
### List ASA Groups a specific ASA User is a member of

<ApiOperation method="GET" url="https://app.scaleft.com/v1/teams/${team_name}/users/${user_name}/groups" />

> This endpoint lists the ASA Groups a specific ASA User belongs to.

Reviewer: ```suggestion
Lists the ASA Groups that a specific ASA User belongs to
```

> This endpoint requires one of the following roles: `access_admin`, `reporting_user`, `access_user`.

Reviewer: ```suggestion
This endpoint requires one of the following roles: `access_admin`, `reporting_user`, or `access_user`.
```
